### ISSUE TITLE
Install does not work

### ISSUE THREAD

**Issue by pjominet** [microsoft/FluentUIEditableDetailsList#51]
Hello yall,
We tried to install your library in our project, but the install fails due to dependency issues. Every solution we tried so far did not resolve the issue, sometimes they even got worse.

npm log: 

npm i fluentui-editable-grid --save
npm WARN ERESOLVE overriding peer dependency
npm WARN Found: @types/react@17.0.27
npm WARN node_modules/@types/react
npm WARN   peer @types/react@">=16.8.0 <18.0.0" from @fluentui/foundation-legacy@8.1.13
npm WARN   node_modules/@fluentui/foundation-legacy
npm WARN     @fluentui/foundation-legacy@"^8.1.13" from @fluentui/react@8.36.2
npm WARN     node_modules/@fluentui/react
npm WARN   14 more (@fluentui/react, @fluentui/react-focus, ...)
npm WARN
npm WARN Could not resolve dependency:
npm WARN peer @types/react@">=16.8.0 <17.0.0" from office-ui-fabric-react@7.177.2
npm WARN node_modules/office-ui-fabric-react
npm WARN   office-ui-fabric-react@"^7.0.0" from fluentui-editable-grid@1.2.1
npm WARN   node_modules/fluentui-editable-grid
npm WARN ERESOLVE overriding peer dependency
npm WARN Found: @types/react-dom@17.0.9
npm WARN node_modules/@types/react-dom
npm WARN   peer @types/react-dom@">=16.8.0 <18.0.0" from @fluentui/react@8.36.2
npm WARN   node_modules/@fluentui/react
npm WARN     @fluentui/react@"^8.30.3" from the root project
npm WARN     1 more (fluentui-editable-grid)
npm WARN   2 more (the root project, fluentui-editable-grid)
npm WARN
npm WARN Could not resolve dependency:
npm WARN peer @types/react-dom@">=16.8.0 <17.0.0" from office-ui-fabric-react@7.177.2
npm WARN node_modules/office-ui-fabric-react
npm WARN   office-ui-fabric-react@"^7.0.0" from fluentui-editable-grid@1.2.1
npm WARN   node_modules/fluentui-editable-grid
npm WARN ERESOLVE overriding peer dependency
npm WARN Found: react@17.0.2
npm WARN node_modules/react
npm WARN   peer react@">=16.8.0 <18.0.0" from @fluentui/foundation-legacy@8.1.13
npm WARN   node_modules/@fluentui/foundation-legacy
npm WARN     @fluentui/foundation-legacy@"^8.1.13" from @fluentui/react@8.36.2
npm WARN     node_modules/@fluentui/react
npm WARN   18 more (@fluentui/react, @fluentui/react-focus, ...)
npm WARN
npm WARN Could not resolve dependency:
npm WARN peer react@">=16.8.0 <17.0.0" from office-ui-fabric-react@7.177.2
npm WARN node_modules/office-ui-fabric-react
npm WARN   office-ui-fabric-react@"^7.0.0" from fluentui-editable-grid@1.2.1
npm WARN   node_modules/fluentui-editable-grid
npm WARN ERESOLVE overriding peer dependency
npm WARN Found: react-dom@17.0.2
npm WARN node_modules/react-dom
npm WARN   peer react-dom@">=16.8.0 <18.0.0" from @fluentui/react@8.36.2
npm WARN   node_modules/@fluentui/react
npm WARN     @fluentui/react@"^8.30.3" from the root project
npm WARN     1 more (fluentui-editable-grid)
npm WARN   5 more (@testing-library/react, styled-components, ...)
npm WARN
npm WARN Could not resolve dependency:
npm WARN peer react-dom@">=16.8.0 <17.0.0" from office-ui-fabric-react@7.177.2
npm WARN node_modules/office-ui-fabric-react
npm WARN   office-ui-fabric-react@"^7.0.0" from fluentui-editable-grid@1.2.1
npm WARN   node_modules/fluentui-editable-grid
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! Found: @types/react@17.0.27
npm ERR! node_modules/@types/react
npm ERR!   peer @types/react@">=16.8.0 <18.0.0" from @fluentui/foundation-legacy@8.1.13
npm ERR!   node_modules/@fluentui/foundation-legacy
npm ERR!     @fluentui/foundation-legacy@"^8.1.13" from @fluentui/react@8.36.2
npm ERR!     node_modules/@fluentui/react
npm ERR!       @fluentui/react@"^8.30.3" from the root project
npm ERR!       1 more (fluentui-editable-grid)
npm ERR!   peer @types/react@">=16.8.0 <18.0.0" from @fluentui/react@8.36.2
npm ERR!   node_modules/@fluentui/react
npm ERR!     @fluentui/react@"^8.30.3" from the root project
npm ERR!     @fluentui/react@"^8.26.0" from fluentui-editable-grid@1.2.1
npm ERR!     node_modules/fluentui-editable-grid
npm ERR!       fluentui-editable-grid@"*" from the root project
npm ERR!   13 more (@fluentui/react-focus, @fluentui/react-hooks, ...)
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer @types/react@">=16.8.0 <17.0.0" from @fluentui/react-focus@7.18.0
npm ERR! node_modules/office-ui-fabric-react/node_modules/@fluentui/react-focus
npm ERR!   @fluentui/react-focus@"^7.18.0" from office-ui-fabric-react@7.177.2
npm ERR!   node_modules/office-ui-fabric-react
npm ERR!     office-ui-fabric-react@"^7.0.0" from fluentui-editable-grid@1.2.1
npm ERR!     node_modules/fluentui-editable-grid
npm ERR!       fluentui-editable-grid@"*" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.

our package.json: 

{
  "name": "app.ui",
  "version": "0.1.0",
  "private": true,
  "proxy": "https://localhost:7011",
  "dependencies": {
    "@azure/msal-browser": "^2.16.1",
    "@fluentui/react": "^8.30.3",
    "@testing-library/jest-dom": "^5.14.1",
    "@testing-library/react": "^12.0.0",
    "@testing-library/user-event": "^13.2.1",
    "@types/jest": "^27.0.1",
    "@types/node": "^16.7.10",
    "@types/react": "^17.0.20",
    "@types/react-dom": "^17.0.9",
    "@types/react-json-tree": "^0.13.0",
    "@types/react-router-dom": "^5.1.8",
    "@types/styled-components": "^5.1.14",
    "react": "^17.0.2",
    "react-dom": "^17.0.2",
    "react-hook-form": "^7.15.3",
    "react-json-tree": "^0.15.0",
    "react-router-dom": "^5.3.0",
    "react-scripts": "4.0.3",
    "rxjs": "^7.3.0",
    "styled-components": "^5.3.1",
    "typescript": "^4.3.4",
    "web-vitals": "^2.1.0"
  },
  "scripts": {
    "start": "set PORT=3333&&react-scripts start",
    "prestart": "copy .\\cert\\cert.pem .\\node_modules\\webpack-dev-server\\ssl",
    "build": "react-scripts build",
    "test": "react-scripts test",
    "eject": "react-scripts eject",
    "csp": "node csp.js",
    "format": "prettier --write src/**/*.ts{,x}"
  },
  "eslintConfig": {
    "extends": [
      "react-app",
      "react-app/jest"
    ]
  },
  "husky": {
    "hooks": {
      "pre-commit": "npm run format && npm audit fix"
    }
  },
  "browserslist": {
    "production": [
      ">0.2%",
      "not dead",
      "not op_mini all"
    ],
    "development": [
      "last 1 chrome version",
      "last 1 firefox version",
      "last 1 safari version"
    ]
  },
  "devDependencies": {
    "eslint-config-prettier": "^8.3.0",
    "eslint-plugin-prettier": "^4.0.0",
    "husky": "^7.0.2",
    "prettier": "^2.3.2"
  }
}

Do you guys know what is wrong, or why it won't install? 
Kind regards

**Comment by YumeEmanuel** [comment]
Hello i'm workin with @pjominet and I'm the front-end developer of our project.

I would like to continue the discussion by adding more information:

When doing the following steps:

1. - npx create-react-app my-app --template @fluentui/cra-template
2. - cd my-app
3. - replaced package.json content with the content provided above
4. - npm install
5. - npm i fluentui-editable-grid --save
6. - It works but there are missing depencencies that your project requires such as @uifabric/react-hooks

After looking at the node_modules I can see there are the following missing dependencies:
1. @uifabric/foundation
2. @uifabric/react-hooks
3. @uifabric/utilities

Maybe there are more missing dependencies but this ones are linked to your library :

For instance
./node_modules/fluentui-editable-grid/dist/editablegrid/columnfiltercallout/filtercallout.jsModule not found: Can't resolve '@uifabric/react-hooks' in 'C:\Users\Emanuel\source\repos\kwhtt-ui\node_modules\fluentui-editable-grid\dist\editablegrid\columnfiltercallout'

asks the @uifabric/react-hooks dependency

By adding manually those dependencies in the node_modules, the lib seems to work BUT there are some warnings/ 'debugger;' in your code (in your editablegridinitialize.js for instance, and there are more)/ console.log('Backup Grid Data'); (in your editablegrid.js (l.37) & helper.js (l.24) for instance that would be nice to remove)

More info about warnings:

1. Warning: Using UNSAFE_componentWillReceiveProps in strict mode is not recommended and may indicate bugs in your code
See https://reactjs.org/link/unsafe-component-lifecycles for details.

* Move data fetching code or side effects to componentDidUpdate.
* If you're updating state whenever props change, refactor your code to use memoization techniques or move it to static getDerivedStateFromProps. Learn more at: https://reactjs.org/link/derived-state

Please update the following components: ModalBase, ResizeGroupBase

2. Warning: Using UNSAFE_componentWillUpdate in strict mode is not recommended and may indicate bugs in your code
See https://reactjs.org/link/unsafe-component-lifecycles for details.

* Move data fetching code or side effects to componentDidUpdate.
Please update the following components: OverflowSetBase

To summarize I have to add manually to node_modules some of @uifabric depencencies, but by doing that I we want to go in production (wih pipelines and so on) we would not get the correct dependencies.

Thanks for your help

**Comment by kushalmehrotra713** [comment]
Also, onRenderItemColumn prop may not work on the wrapper component since the component internally already uses the onRender function to achieve all the customizations that the Grid wrapper component supports. Hence, if you are trying to add any further customizations to columns, you may want to have a look CreateColumnConfigs() function in editablegrid.tsx. This is where all the editability renders reside for this grid. Other approach, as you mentioned, is to use onRenderRow prop to achieve the desired functionality.
If I understand correctly, you are trying to wrap some of your column values with a <div>. If that is the case, I can probably expose some functionality for a user to be able to do that via onRender props of IColumnConfig. In such a case, it would override the inbuilt Wrapper component's onRender and you can plug in whatever you want. Let me know if this sounds useful to you.

**Comment by kushalmehrotra713** [comment]
Hi @YumeEmanuel 
I have updated the npm to support onRender in column declarations itself(IColumnConfig). This will enable you to provide your own render implementation for whichever columns you wish. 
However, please note that if you require editability features on such a column, you will have to provide your own custom implementation, since this column will now be rendered based on your implementation.

As for selecting the grid rows, will it help if I provide a property that takes the row indices as input and selects the same rows everytime you provide the indices? That will be done without having to rerender the grid component.

**Comment by YumeEmanuel** [comment]
Hello @kushalmehrotra713 , 

For the own implementation I understand your point (of course your implementation for editable column needs to be still here, but of coruse I I want to provide something else for it, you new feature will be good: and I udnerstand the consequences).

Concerning the feature to have a method or something to give (row indices) to the component will be super yes.

PS: I need to tell you I'm using another library (which has nothing to do with fluent-ui) because I need complicated features and that's why I'm not currently using your library now. But your library will help other people if they want something light and integrated with fluent-ui. Thank you for your time.

**Comment by kushalmehrotra713** [comment]
Sure @YumeEmanuel . Closing this issue now.
